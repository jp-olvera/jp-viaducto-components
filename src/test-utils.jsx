import React from 'react';
import { render } from '@testing-library/react';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }) => {
  return <ConfigProvider>{children}</ConfigProvider>;
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
