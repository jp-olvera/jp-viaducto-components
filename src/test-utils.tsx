import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }: any) => (
  <ConfigProvider>{children}</ConfigProvider>
);

jest.mock('./cells/Input/amex.svg', () => null);
jest.mock('./cells/Input/mastercard.svg', () => null);
jest.mock('./cells/Input/visa.svg', () => null);
jest.mock('./cells/Dropdown/sorting.svg', () => null);
jest.mock('./cells/BareButton/close.svg', () => null);

const customRender = (ui, options?) => render(ui, { wrapper: AllTheProviders, ...options });
// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
