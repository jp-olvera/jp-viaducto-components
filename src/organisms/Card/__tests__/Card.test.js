import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '../../../test-utils';
import { Card } from '..';

jest.mock('../../../cells/Dropdown/sorting.svg', () => null);

describe('<Card/>', () => {
  test('should be visible', () => {
    render(
      <Card
        collapsible={false}
        collapse={false}
        src="https://homepages.cae.wisc.edu/~ece533/images/girl.png"
        content={<h1>Body</h1>}
        footer={<h2>Footer</h2>}
        onlyImage={false}
      />,
    );
    expect(screen.queryByText('45 min')).toBeInTheDocument();
  });
  test('render Card collapsible (large)', () => {
    render(
      <Card
        collapsible
        collapse={false}
        src="https://homepages.cae.wisc.edu/~ece533/images/girl.png"
        content={<h1>Body</h1>}
        footer={<h2>Footer</h2>}
        onlyImage={false}
      />,
    );
    expect(screen.queryByText('Designer')).toBeInTheDocument();
  });
});
