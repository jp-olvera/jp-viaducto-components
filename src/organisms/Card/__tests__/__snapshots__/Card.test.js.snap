// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> render Card with default props and create snapshot 1`] = `
<div>
  <div
    class="StyledCard__StyledNormalCard-sc-15ny05r-1 jCVood"
    elevation="1"
  />
</div>
`;
